cptv-upload.py

```diff
@@ -0,0 +1,125 @@
+from deviceapi import DeviceAPI
+
+import argparse
+
+import os
+
+import json
+
+class CPTVUploader:
+    def __init__(self):
+        self.url = None        
+        self.source_dir = None
+        self.device_name = None
+        self.device_password = None
+
+    def process(self):
+        print('Uploading to  ', self.url)
+        api = DeviceAPI(self.url, devicename = self.device_name, password = self.device_password)
+
+        files = self._find_files_to_upload()
+        for file in files:
+            self._uploadfile(api, file)
+
+    def _uploadfile(self, api, filename):
+        props = self._readPropertiesFromFile(filename)
+
+        api.uploadrecording(filename, props)
+
+
+    def _uploader(self, queue, api):
+        while True:
+            
+            filename = queue.get()
+
+            if filename is None:
+                break
+
+            try:
+                _uploadfile(api, filename)
+
+            finally:
+                queue.task_done()
+
+    def _find_files_to_upload(self):
+        cptvfiles = list()
+
+        for root, dirs, files in os.walk(self.source_dir):
+            for file in files:
+                if file.lower().endswith(".cptv"):
+                    cptvfiles.append(os.path.join(root, file))
+        
+        return cptvfiles
+
+    def _readPropertiesFromFile(self, filename): 
+        basefile = os.path.splitext(filename)[0]
+        jsonfilename = basefile + '.txt'
+
+        if (os.path.isfile(jsonfilename)):
+            with open(jsonfilename, 'r') as propsfile: 
+
+                oldprops = json.load(propsfile)
+
+                # List of properties to transfer.   Many such as Id, we don't want to transfer. 
+                propTypesToTransfer = ("batteryCharging", "additionalMetadata", "comment", "location", 
+                            "fileSize", "batteryLevel", "duration", "rawFileSize", "airplaneModeOn", 
+                            "version", "recordingDateTime", "fileMimeType", "type")
+
+                newProps = dict()
+
+                for key in propTypesToTransfer: 
+                    if (key in oldprops and oldprops[key] is not None):
+                        newProps[key] = oldprops[key]
```

Comment:
Another way to do this (for educational purposes, what you did is fine too):

```python
# Note: propTypesToTransfer is now a set (see the the braces). Lookup into a set is O(1).
propTypesToTransfer = {"batteryCharging", "additionalMetadata", "comment", "location", 
                            "fileSize", "batteryLevel", "duration", "rawFileSize", "airplaneModeOn", 
                            "version", "recordingDateTime", "fileMimeType", "type"}
newProps = dict((k, v) for k, v in oldprops.items() if k in propTypesToTransfer)
```
If you want to know how the above works search for "python generator expressions".